### ISSUE TITLE
Convert wercker regressions that use maven-checkstyle-plugin to CLI based

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#5386]
See https://issues.apache.org/jira/browse/MCHECKSTYLE-346 .
We need to deprecate all usage of `maven-checkstyle-plugin` so we can start implementing backward breaking changes.

Everything in wercker uses `maven-checkstyle-plugin` to run checkstyle on the projects and will have to be converted to using the raw CLI.
See https://github.com/checkstyle/checkstyle/blob/master/wercker.yml

This may be difficult for some projects as they use custom dependencies. We may have to start discussions with the project owners on how we can best do this and disable them until we can get things worked out.

**Comment by romani** [comment]
> Everything in wercker uses maven-checkstyle-plugin

here is nuance, all "no-error" builds are execution of how projects owners do validation in their CIs.
Plugin usage should stay as is :(. in other case it will be nightmare to keep them in sync and override smth in your configuration.

"no-exception" builds are where we can start to use our new abilities of regression tool.

> We may have to start discussions with the project owners on how we can best do this and disable them until we can get things worked out.

as we already see, they merge quickly all updates from us to their projects. So if we prepare update to them, they will merge it.

**Comment by rnveach** [comment]
> Plugin usage should stay as is :(. in other case it will be nightmare to keep them in sync and override smth in your configuration.

@romani If we do not drop all instances of `maven-checkstyle-plugin` we could never implement backward breaking changes as plugin will always fail and CI will not be fixed until new version of plugin is released. Migrating a few is not an option if we want to continue to use this CI. We must use CLI (or regression) for all runs.
To fully break away from `maven-checkstyle-plugin` we can't use it anywhere and only use what we created.

> in other case it will be nightmare to keep them in sync and override smth in your configuration.

If you are referring to keeping CLI connected to their configuration, most people don't move their configuration often. If it becomes a big issue, we will have to talk to project owners and come up with a different way.

**Comment by romani** [comment]
> Migrating a few is not an option if we want to continue to use this CI. We must use CLI (or regression) for all runs.

if change is such a huge and damaging we can postpone its merge till date of release, and disable affected projects in CI for some time.
I have no doubt that we will not be able to properly keep suppressions and one day we will loose what is critical for users.
What we use now, is we test our changes by users acceptance. One day we will send PR and owners will reject it as not appropriate for them.
We check acceptance by users(not checksyle owners) on early steps.

**Comment by rnveach** [comment]
@romani 
If we can disable wercker for breaking changes, then we might as well disable sevntu and our regression tool too as wercker is just another form of regression.
Waiting on a fix from the maven plugin, or disabling CI, is the same thing we have been doing already for more than 2 years and it has not gotten us anything. This is how long we have been waiting for a maven release. Not converting **everything** to use our CLI is just continuing that tradition.
I feel it is time for a change and to try something new. We need to convert **everything** to something we own which is the CLI. We should not be held back from releasing changes because of an external entity. We get around this for wercker right now by applying our own patches on external projects. Constantly disabling and re-enabling wercker is just a patch and not a solution. Disabling it for 2 years waiting on a maven plugin release isn't a solution either.
We have around 16 other deprecated methods/constructors that we can remove once we break away from the maven plugin. Dancing back and forth to getting those 16 removed will just complicate the process even more.

If we are not going to change everything, then we might as well forget all the issues I have opened for migrating away from maven plugin as we will still be locked into using them and will never be free to do as we wish without them.

It also seems to me we are more concerned with this one utility more than others. We have broken eclipse-cs multiple times, but we have never once ingrained them into us or at the very least put some type of test for it in wercker. Same goes with sonar. There is no consistency and this is just more reason why we should break away from the maven plugin.

**Comment by romani** [comment]
> I feel it is time for a change and to try something new. 

yes we come to last point where we live without changes. Unfortunately maven projects were easy to use in CI.

> We should not be held back from releasing changes because of an external entity. 

ideally - yes. This level of regression testing was convenient and easy to archive.

> Dancing back and forth to getting those 16 removed will just complicate the process even more.

no arguing here.

> I only started all this because I thought you were in agreement that we shouldn't be dependent on the plugin like this

I agree with you .... but in my experience ..... I spent a lot of time to keep CI green, problems happens as CIs projects are changing each day and .... . So I tried to use/find projects that restrict checkstyle by build so any new violation is clear signal of regression and how user will experience it.
One more point that I tried to archive is to work on their latest sources(that helps us to avoid using our forks of projects in CI ....) .... it is good point. But this point is not that big as moving on with braking changes ...
Additional advantage is that we see affect on plugins before release... but is not a reason to be restrained by some plugins.

So for some time ... we can enforce validation on current hash commits. Create suppressions if required for CLI usage. If any changes be required , we will make fork and use forked repos in our CIs.

> We have broken eclipse-cs multiple times, but we have never once ingrained them into us or at the very least put some type of test for it in wercker.

consistency cost a lot of time ... I have no time to afford this for all major plugins. I need your help to make releases.

====================

Summary: 
I am completely ok to remove maven plugin usage from regression tool completely.
I am ok temporary (till maven plugin updated) to convert CI "no-error"/"no-exceptions" projects to covert to CLI. Ones plugin become up to date, it is better to come back to use to latest projects code to avoid forks.
Please help me with CIs resolving.

**Comment by Rahulkhinchi03** [comment]
@romani @rnveach does this issue still valid?
As per https://github.com/checkstyle/checkstyle/issues/11564, The Wercker services are going to be shut down, so?

**Comment by romani** [comment]
Yes, valid as all jobs are just moved to other CI services

**Comment by romani** [comment]
@rnveach , we might create new issue, as this issue with better instructions as this becomes too confusing. Please create new issue.

**Comment by rnveach** [comment]
Closed as new issue created at #11602
